### PR TITLE
update CellEditor components to ES6

### PR DESCRIPTION
**Pull request by Hamms**
and make them all inherit from a single parent class.

The diffs for this change are going to be fairly difficult to parse; it'll probably be easier to actually look at before/afters of the components in their entirety.

Most notably, CellEditor ([before](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/components/CellEditor.jsx), [after](https://github.com/code-dot-org/code-dot-org/blob/ec20c688ec529496483eebf72390f69fcb1f4ad2/apps/src/code-studio/components/CellEditor.jsx)) has been restructured into a class with a handful of discrete, overriddable methods, and the other cell editor classes have been refactored to extend from that class.

This usually involves simply overriding/extending the `renderFields` and `getSelectFieldNames` methods.

Changes:

BeeCellEditor.jsx
CellEditor.jsx
GridEditor.jsx
HarvesterCellEditor.jsx
PlanterCellEditor.jsx
StarWarsGridCellEditor.jsx
StudioCellEditor.jsx

Before:

[BeeCellEditor.jsx](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/components/BeeCellEditor.jsx)
[CellEditor.jsx](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/components/CellEditor.jsx)
[GridEditor.jsx](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/components/GridEditor.jsx)
[HarvesterCellEditor.jsx](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/components/HarvesterCellEditor.jsx)
[PlanterCellEditor.jsx](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/components/PlanterCellEditor.jsx)
[StarWarsGridCellEditor.jsx](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/components/StarWarsGridCellEditor.jsx)
[StudioCellEditor.jsx](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/components/StudioCellEditor.jsx)

After:

[BeeCellEditor.jsx](https://github.com/code-dot-org/code-dot-org/blob/ec20c688ec529496483eebf72390f69fcb1f4ad2/apps/src/code-studio/components/BeeCellEditor.jsx)
[CellEditor.jsx](https://github.com/code-dot-org/code-dot-org/blob/ec20c688ec529496483eebf72390f69fcb1f4ad2/apps/src/code-studio/components/CellEditor.jsx)
[GridEditor.jsx](https://github.com/code-dot-org/code-dot-org/blob/ec20c688ec529496483eebf72390f69fcb1f4ad2/apps/src/code-studio/components/GridEditor.jsx)
[HarvesterCellEditor.jsx](https://github.com/code-dot-org/code-dot-org/blob/ec20c688ec529496483eebf72390f69fcb1f4ad2/apps/src/code-studio/components/HarvesterCellEditor.jsx)
[PlanterCellEditor.jsx](https://github.com/code-dot-org/code-dot-org/blob/ec20c688ec529496483eebf72390f69fcb1f4ad2/apps/src/code-studio/components/PlanterCellEditor.jsx)
[StarWarsGridCellEditor.jsx](https://github.com/code-dot-org/code-dot-org/blob/ec20c688ec529496483eebf72390f69fcb1f4ad2/apps/src/code-studio/components/StarWarsGridCellEditor.jsx)
[StudioCellEditor.jsx](https://github.com/code-dot-org/code-dot-org/blob/ec20c688ec529496483eebf72390f69fcb1f4ad2/apps/src/code-studio/components/StudioCellEditor.jsx)

### CODE REVIEW

File: apps/src/code-studio/components/BeeCellEditor.jsx [hunk 1]
```diff
@@ -1,43 +1,27 @@
...
+export default class BeeCellEditor extends CellEditor {
   /**
-   * Focus and select the input that contains this cell's serialized
-   * JSON representation on creation or update, to provide a convenient
-   * copy/paste update route.
+   * @override
    */
+  getSelectFieldNames() {
+    return super.getSelectFieldNames().concat([
+      'featureType', 'cloudType', 'flowerColor'
```
Comment (joshlory) [inline apps/src/code-studio/components/BeeCellEditor.jsx:16]: Enum / shared const?

File: apps/src/code-studio/components/CellEditor.jsx [hunk 2]
```diff
@@ -1,62 +1,118 @@
...
 
         <label htmlFor="range">Range (defaults to value):</label>
         <input type="number" name="range" value={values.range} disabled={values.featureType === 'undefined'} onChange={this.handleChange} />
+      </div>
+    );
+  }
+
+  render() {
+    return (
+      <form className="span4 offset1">
```
Comment (joshlory) [inline apps/src/code-studio/components/CellEditor.jsx:109]: Would be good to not rely on external styling (in a future PR!)

File: apps/src/code-studio/components/GridEditor.jsx [hunk 3]
```diff
@@ -162,9 +162,15 @@ var GridEditor = React.createClass({
   /**
    * When a given cell is modified, update the grid
    */
-  handleCellChange: function (newSerializedCell) {
+  handleCellChange: function (newSerializedCellData) {
     var row = this.state.selectedRow;
     var col = this.state.selectedCol;
+
+    const oldCell = this.state.cells[row][col].serialize();
+    console.log(oldCell, newSerializedCellData);
```
Comment (joshlory) [inline apps/src/code-studio/components/GridEditor.jsx]: Extra logging?

File: apps/src/code-studio/components/StarWarsGridCellEditor.jsx [hunk 4]
```diff
@@ -18,47 +19,41 @@ const options = {
...
       <form className="span4 offset1">
-        <img src={tileGuide}/>
+        <img src={tileGuide} />
         <header>
           <strong>Editing Cell ({this.props.row}, {this.props.col})</strong>
         </header>
 
         <label htmlFor="tileType">Tile Type (required):</label>
-        <select ref="type" name="tileType" value={type} onChange={this.handleChange}>
+        <select ref={c => { this.type = c; }} name="tileType" value={type} onChange={this.handleChange}>
```
Comment (joshlory) [inline apps/src/code-studio/components/StarWarsGridCellEditor.jsx:48]: What does the `ref` arrow function here do?

File: apps/src/code-studio/components/StudioCellEditor.jsx [hunk 5]
```diff
@@ -1,75 +1,52 @@
...
-      if (values[value] === undefined) {
-        values[value] = 'undefined';
-      }
-    });
+  /**
+   * @override
+   */
+  getSelectFieldNames() {
+    return super.getSelectFieldNames().concat([
+      'speed', 'size', 'direction', 'emotion', 'sprite'
```
Comment (joshlory) [inline apps/src/code-studio/components/StudioCellEditor.jsx:39]: Enum / shared const?

File: apps/src/code-studio/components/StarWarsGridCellEditor.jsx (hunk 4 above)
Comment (Hamms) [reply apps/src/code-studio/components/StarWarsGridCellEditor.jsx:48]: It's the new preferred way of saving references; the old string-based refs are considered legacy: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-string-refs.md#prevent-using-string-references-reactno-string-refs

https://facebook.github.io/react/docs/refs-and-the-dom.html#legacy-api-string-refs

Comment (joshlory) [reply apps/src/code-studio/components/StarWarsGridCellEditor.jsx:48]: TIL!

**Comment by joshlory** [comment]
CircleCI UI test failures appear to be a [temporary Firebase outage](https://codedotorg.slack.com/archives/C2ALWLRHN/p1498669190892639), unrelated to your change.

**Pull request merged**